src/realm/table.hpp

```diff
@@ -1202,6 +1203,17 @@ inline bool Table::is_group_level() const noexcept
     return bool(get_parent_group());
 }
 
+inline Obj Table::operator[](Mixed pk) const
```

Comment:
I think this function is somewhat awkward. It looks like you want an object with a primary key, but if there is no primary key column, then you hope that pk contains an ObjKey. I really don't think this is needed.

Reply:
It isn't needed, of course, but it reduces tedious boilerplate in unit tests. The Sync unit tests are suffering from this especially, because `ObjKey` is not stable between clients, so we do a lot of lookups by primary key.

Reply:
I think the proper place for something like this would be some kind of supportive function in the Sync unit tests then.

Reply:
I would be in favor of cleaning up the `Table` API with a "convenience" wrapper that accepts string names for columns and other patterns with suboptimal performance. I believe that unit tests in Core would benefit from that as well, as well as making the `Table` API easier to understand and use correctly.

It seems to me there is enough resistance towards this change, so I will revert it.